src/main/java/com/avioconsulting/log4j/sqs/processor/FifoMessageProcessor.java

```diff
@@ -21,55 +21,54 @@
-			sendMessageRequest.setMessageBody(String.format("currentPart=%d|totalParts=%d|uuid=%s|message=%s", i + 1, splitMessage.length, uuid, splitMessage[i]));
-			sendMessageRequest.setMessageGroupId(uuid.toString());
-			sendMessageRequest.setQueueUrl(processorAttributes.getQueueUrl());
-			sendMessageRequestList.add(sendMessageRequest);
-			messageRequestWrapper.setSendMessageRequest(sendMessageRequestList);
-		}
-		return messageRequestWrapper;
-	}
+        for (int i = 0; i < splitMessage.length; i++) {
+            logger.debug("Sending message {} of {}", i + 1, splitMessage.length);
+            SendMessageRequest sendMessageRequest = new SendMessageRequest();
+            sendMessageRequest.setMessageBody(String.format("currentPart=%d|totalParts=%d|uuid=%s|message=%s", i + 1, splitMessage.length, uuid, splitMessage[i]));
+            sendMessageRequest.setMessageGroupId(uuid.toString());
+            sendMessageRequestList.add(sendMessageRequest);
+            messageRequestWrapper.setSendMessageRequest(sendMessageRequestList);
+        }
+        return messageRequestWrapper;
+    }
 
-	/**
-	 * Split the input string into a list of strings, each with a maximum memory size of maxsize bytes.
-	 *
-	 * @param src the source string
-	 * @param encoding the encoding to use for the output string
-	 * @param maxsize the maximum size of the string parts in bytes
-	 * @return an array of strings, each with a maximum length of maxsize bytes
-	 */
-	protected static String[] splitStringByByteLength(final String src, final String encoding, final int maxsize) {
-		Charset cs = Charset.forName(encoding);
-		CharsetEncoder coder = cs.newEncoder();
-		ByteBuffer out = ByteBuffer.allocate(maxsize);  // output buffer of required size
-		CharBuffer in = CharBuffer.wrap(src);
-		List<String> ss = new ArrayList<>();            // a list to store the chunks
-		int pos = 0;
-		while (true) {
-			CoderResult result = coder.encode(in, out, true); // try to encode as much as possible
-			int newpos = src.length() - in.length();
-			String s = src.substring(pos, newpos);
-			ss.add(s);                                  // add what has been encoded to the list
-			pos = newpos;                               // store new input position
-			out.rewind();                               // and rewind output buffer
-			if (!result.isOverflow()) {
-				break;                                  // everything has been encoded
-			}
-		}
-		return ss.toArray(new String[0]);
-	}
+    /**
+     * Split the input string into a list of strings, each with a maximum memory size of maxsize bytes.
+     *
+     * @param src      the source string
+     * @param encoding the encoding to use for the output string
+     * @param maxsize  the maximum size of the string parts in bytes
+     * @return an array of strings, each with a maximum length of maxsize bytes
+     */
+    public static String[] splitStringByByteLength(final String src, final String encoding, final int maxsize) {
```

Comment:
Was there a reason for changing this to public?  Nevermind I see it was for testing purposes.  We probably should remove the static though, this is used from an instance method only, no reason for it to be static.

Reply:
Done